shared/AppInsightsCore/src/JavaScriptSDK/CoreUtils.ts

```diff
@@ -146,6 +183,63 @@ export function objForEachKey(target: any, callbackfn: (name: string, value: any
     }
 }
 
+/**
+ * Effectively assigns all enumerable properties (not just own properties) and functions (including inherited prototype) from 
+ * the source object to the target, it attempts to use proxy getters / setters (if possible) and proxy functions to avoid potential
+ * implementation issues by assigning prototype functions as instance ones
+ * 
+ * This method is the primary method used to "update" the snippet proxy with the ultimate implementations.
+ * 
+ * Special ES3 Notes:
+ * Updates (setting) of direct property values on the target or indirectly on the source object WILL NOT WORK PROPERLY, updates to the 
+ * properties of "referenced" object will work (target.context.newValue = 10 => will be reflected in the source.context as it's the
+ * same object). ES3 Failures: assigning target.myProp = 3 -> Won't change source.myProp = 3, likewise the reverse would also fail.
+ * @param target - The target object to be assigned with the source properties and functions
+ * @param source - The source object which will be assigned / called by setting / calling the targets proxies
+ * @param chkSet - An optional callback to determine whether a specific property/function should be proxied
+ * @memberof Initialization
+ */
+export function proxyAssign(target: any, source: any, chkSet?: (name: string, isFunc?: boolean, source?: any, target?: any) => boolean) {
+    if (target && source && target !== source && _isObject(target) && _isObject(source)) {
+        // effectively apply/proxy full source to the target instance
+        for (const field in source) {
+            if (CoreUtils.isString(field)) {
+                let value = source[field] as any;
+                if (_isFunction(value)) {
+                    if (!chkSet || chkSet(field, true, source, target)) {
+                        // Create a proxy function rather than just copying the (possible) prototype to the new object as an instance function
+                        target[field as string] = (function(funcName: string) {
+                            return function() {
+                                // Capture the original arguments passed to the method
+                                var originalArguments = arguments;
+                                return source[funcName].apply(source, originalArguments);
```

Comment:
Changes in latest version to perform a function name lookup on the object, this supports the changing of the function (at runtime) from a prototype based function to an instance one or just updating / wrapping the function with a subclass or during testing with sinon-spy.